integration/spark/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -167,6 +167,13 @@ object CarbonMergeFilesRDD {
             executorService.submit(new Runnable {
               override def run(): Unit = {
                 ThreadLocalSessionInfo.setCarbonSessionInfo(carbonSessionInfo)
+                val oldFolder = FileFactory.getCarbonFile(
```

Comment:
why this change required now? Is it fixing any bug?

Reply:
then, during clean files these moved index files are taken care to delete ?

Reply:
Yes, during clean files it collects all index files of a segment and whichever are not present in segment file are deleted.